paddle/operators/math/sampler.cc

```diff
@@ -0,0 +1,47 @@
+#include "sampler.h"
+
+namespace paddle {
+namespace random {
+
+Sampler::~Sampler() {}
+
+UniformSampler::UniformSampler(int64 range)
+    : Sampler(range), inv_range_(1.0 / range) {
+  std::random_device r;
+  random_engine_ = std::make_shared<std::mt19937>(r());
+  dist_ = std::make_shared<std::uniform_int_distribution<>>(0, range);
+}
+
+int64 UniformSampler::Sample() const { return (*dist_)(*random_engine_); }
+
+float UniformSampler::Probability(int64 value) const { return inv_range_; }
+
+LogUniformSampler::LogUniformSampler(int64 range)
+    : Sampler(range), log_range_(log(range + 1)) {
+  std::random_device r;
```

Comment:
It is better to provide an option to explicitly set the random seed.

Reply:
Thx.